pkg/infrastructure/infrastructure.go

```diff
@@ -0,0 +1,92 @@
+package infrastructure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "github.com/openshift/api/config/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	infrastructureName = "cluster"
+)
+
+// Builder provides a struct for infrastructure object from the cluster and a infrastructure definition.
+type Builder struct {
+	// infrastructure definition, used to create the infrastructure object.
+	Definition *v1.Infrastructure
+	// Created infrastructure object.
+	Object *v1.Infrastructure
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// Pull loads an existing infrastructure into Builder struct.
+func Pull(apiClient *clients.Settings) (*Builder, error) {
+	glog.V(100).Infof("Pulling existing infrastructure name: %s", infrastructureName)
+
+	builder := Builder{
+		apiClient: apiClient,
+		Definition: &v1.Infrastructure{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: infrastructureName,
+			},
+		},
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("infrastructure object %s doesn't exist", infrastructureName)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Exists checks whether the given infrastructure exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof(
```

Comment:
To have it consistent with the rest of `Infof()`calls, it would be nice to have this in one line. 

Seems to be the only one diverting from the pattern.